STAR/2.4.2a/AlignReads.nf

```diff
@@ -0,0 +1,43 @@
+process AlignReads {
+    tag "${sample}_STAR"
+    publishDir "$params.outdir/$sample/STAR", mode: 'copy'
+    cpus 12
+    penv 'threaded'
+    memory '70 GB'
+    time '1h'
+
+    input:
+    set val(sample), file(r1_fastqs), file(r2_fastqs)
+    file(index)
+
+    output:
+    set val(sample),file ("${sample}.sorted.mdup.bam" )
+    file ("${sample}.sorted.mdup.bam.bai" )
+    file ("${sample}.Log.final.out")
+    file ("${sample}.Log.out")
+    file ("${sample}.SJ.out.tab")
+    file ("${sample}.Log.progress.out") 
+
+    shell:
+    def barcode = r1_fastqs[1].simpleName
+    def r1_fastqs = r1_fastqs.collect{ "$it" }.join(",")
+    def r2_fastqs = r2_fastqs.collect{ "$it" }.join(",")
+
+    """
+    set -o pipefail
```

Comment:
This has to do with how the bash scripts fail? Should we add it to all our modules?